variables.tf

```diff
@@ -46,6 +46,26 @@ variable "burstable_mode" {
   default     = null
 }
 
+variable "market_type" {
+  type        = string
+  description = "(Optional) Type of market for the instance. Valid values are `spot` and `capacity-block`. If the value set is `spot`, the `spot_options_attributes` variable becomes adjustable."
```

Comment:
You may consider custom validation here https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules to ensure only `spot` or `capacity-block` is used

Reply:
will do, thanks @nitrocode

Reply:
updated https://github.com/cloudposse/terraform-aws-ec2-instance/pull/202/commits/c18468768c919ad5a9b75338b3e9cd0e7b7dadeb